### ISSUE TITLE
Official website

### ISSUE THREAD

**Issue by OxygenCobalt** [OxygenCobalt/Auxio#299]
Make an official website for Auxio once the major features like #36 are done. This is largely an excuse to force myself to learn more about website development.

**Comment by e-zk** [comment]
Any initial thoughts on how the website will be constructed? (I realise this probably isn't a priority ATM, just curious).

I'm definitely one for static or near-static HTML sites, just personally. I'm quite fluent in HTML+CSS so if you're thinking of going the simple static route w/o any CSS frameworks I can help out with CSS.

**Comment by OxygenCobalt** [comment]
Yeah, it's going to be largely static but perhaps with some interesting visuals. This is mostly a personal project in that regard, as I am not very familiar with website development at all (Just look at [my website](oxycblt.org)).

**Comment by ChristianCiach** [comment]
I am a backend developer, so I cannot really offer any qualified recommendations, but..

Some of the frontend developers of the company I am working for have recommended https://gohugo.io to me. Just grab one of the [themes](https://themes.gohugo.io)  (or build your own) and you can use Hugo to render a bunch of markdown files into a static website. 

My plan is to use Hugo from inside the CI pipeline to render the markdown files under version control and push the output to Gitlab pages. But I didn't have any time to try this yet, so take my recommendation with a grain of salt.

**Comment by OxygenCobalt** [comment]
Interesting. In an ideal world, I would want a well-themed front page, but with other pages being rendered markdown. This would include the wiki as well, as I can theoretically grab the repository and then use the markdown files.

**Comment by KraXen72** [comment]
there's a bazillion solutions to generate static sites nowadays, hugo, jekyll, elder.js, astro, 11ty, etc.
ive had pretty good success with vitepress,  you can provide a flashy homepage in vue and then the rest is rendered in markdown, with a nice theme, quick search (once you publish it you can get a free api key for algolia docsearch) and a client side router, which means that all subpage navigations are instantaneous, no full refreshing. it also can be published on GitHub pages (free hosting)
only caveats are:
- it's currently in alpha. it does meet the requitements you listed to 100% though.
- the sidebar (list of all md pages in folders as categories) does not automatically generate - i have written a script to do that though, it's in my tech-support-wiki repo, and I'd be willing to adjust it for your needs on auxios page.
- if you don't know vue and want something even flashier than their main site uses (vitepress.vuejs.org), youll have to get someone who knows vue, or learn it's basics (pretty easy framework)

**Comment by OxygenCobalt** [comment]
Vitepress seems like a good plan. If I remember correctly, vue is a pretty lightweight framework compared to the likes of React, so it seems like a good fit for a "Largely static but with neat animations" type website.

**Comment by satisfiedsingle** [comment]
Hi I would like to suggest something Not from a development angle. Please try to get this domain if you can it looks pretty cool- auxio.app

**Comment by OxygenCobalt** [comment]
Seems like a good idea. I'll yoink it while it's still cheap.
